onnxruntime/core/providers/cpu/tensor/onehot.cc

```diff
@@ -28,29 +28,54 @@ namespace onnxruntime {
 // spec: https://github.com/onnx/onnx/blob/master/docs/Operators.md#OneHot
 
 // T1: indices, T2: depth, T3: values
-#define REG_TYPED_ONE_HOT_OP(types_str, in_type, out_type, depth_type)     \
```

Comment:
Concatenate the content of REG_TYPED_ONE_HOT_OP_V9_10 and REG_TYPED_ONE_HOT_OP_V11 into REG_TYPED_ONE_HOT_OP could reduce line of changes.

Reply:
negative axis is already supported previously and already have test case covered.